solo-io/gloo
# Fully default gwparams

# Description

This PR simplifies the way default values are derived when deploying a dynamically generated gateway proxy from the Kube Gateway API. Previously, there were some defaults defined in `projects/gateway2/deployer/values.go` which were not user-facing. With the introduction of a default GatewayParameters resource, we have the ability to move all user-facing default values into the main Gloo Gateway helm chart at `install/helm/gloo/values-template.yaml`. 

This PR moves all default values in order that they may be configurable by the user.
## API changes
- Autoscaling has been removed from GatewayParameters CR since it was non-functional and will need some reworking later anyway. 

## Code changes
- Removed code around translating/merging/setting autoscaling in the deployer and in templates

## CI changes
- Added `^TestK8sGatewayNoDefaultGatewayParameters$` to cluster_two of the Kubernetes End-to-End tests.
  - cluster_two was chosen based on having the lowest runtime in recent successful CI runs

# Context
In order to run on OpenShift, we have to be able to modify/unset the security contexts. Before this work, the default values for the envoy container's security context were in the internal helm chart and inaccessible to the user. Because of the inheritance paradigm selected for GatewayParameters (i.e. nil value == inherit from parent), it was not possible to unset a value that had a default. By making the defaults user-facing, users can set the default to `null` via helm and achieve the desired effect.

## Interesting decisions
TODO

## Testing steps
TODO
1. Install helm with `--set kubeGateway.gatewayParameters.glooGateway.envoyContainer.securityContext=null`
2. apply a gateway
3. see that deployed proxy has no container security context

## Notes for reviewers
TODO

# Checklist:
